functions/Export-DbaDiagnosticQuery.ps1

```diff
@@ -0,0 +1,140 @@
+function Export-DbaDiagnosticQuery {
+<#
+.SYNOPSIS 
+Export-DbaDiagnosticQuery can convert ouput generated by Invoke-DbaDiagnosticQuery to CSV or Excel
+
+.DESCRIPTION
+The default output format of Invoke-DbaDiagnosticQuery is a custom object. It can also output to CSV and Excel. 
+However, CSV output can generate a lot of files and Excel output depends on the ImportExcel module by Doug Fike (https://github.com/dfinke/ImportExcel)
+Export-DbaDiagnosticQuery can be used to convert from the default export type to the other available export types.
+
+.PARAMETER InputObject
+Specifies the objects to convert
+	
+.PARAMETER ConvertTo
+Specifies the output type. Valid choices are Excel and CSV. CSV is the default.
+	
+.PARAMETER Path
+Specifies the path to the output files. 
+	
+.PARAMETER Silent
+Use this switch to disable any kind of Output messages
+
+.NOTES
+Author: André Kamman (@AndreKamman), http://clouddba.io
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Export-DbaDiagnosticQuery
+
+.EXAMPLE  
+Invoke-DbaDiagnosticQuery -SqlInstance sql2016 | Export-DbaDiagnosticQuery -Path c:\temp
+
+Converts output from Invoke-DbaDiagnosticQuery to multiple CSV files
+	
+.EXAMPLE 
+$output = Invoke-DbaDiagnosticQuery -SqlInstance sql2016
+Export-DbaDiagnosticQuery -InputObject $output -ConvertTo Excel
+
+Converts output from Invoke-DbaDiagnosticQuery to Excel worksheet(s) in the Documents folder
+
+#>
+	[CmdletBinding()]
+	param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object[]]$InputObject,
+		[ValidateSet("Excel", "Csv")]
+		[string]$ConvertTo = "Csv",
+		[System.IO.FileInfo]$Path = [Environment]::GetFolderPath("mydocuments"),
+		[string]$Suffix = "$(Get-Date -format 'yyyyMMddHHmmssms')",
+		[switch]$Silent
+	)
+	
+	begin {
+		if ($convertto -eq "Excel") {
```

Comment:
Recommend: Match variable casing to parameter name, so it is readily visible that the variable is an original parameter